server/src/main/java/org/elasticsearch/action/admin/indices/template/get/GetIndexTemplatesResponse.java

```diff
@@ -65,7 +68,12 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
         builder.startObject();
         for (IndexTemplateMetadata indexTemplateMetadata : getIndexTemplates()) {
-            IndexTemplateMetadata.Builder.toXContent(indexTemplateMetadata, builder, params);
+            if(builder.getRestApiVersion() == RestApiVersion.V_7 &&
+                params.paramAsBoolean(INCLUDE_TYPE_NAME_PARAMETER, DEFAULT_INCLUDE_TYPE_NAME_POLICY)) {
```

Comment:
I think I'd prefer an explicit `false` as the default value here to make it clearer that the normal path is to not output types.